File: sdkwork-app-sdk-go/api/documents.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/app-sdk/types"
    sdkhttp "github.com/sdkwork/app-sdk/http"
)

type DocumentsApi struct {
    client *sdkhttp.Client
}

func NewDocumentsApi(client *sdkhttp.Client) *DocumentsApi {
    return &DocumentsApi{client: client}
}

// Get document detail
func (a *DocumentsApi) GetDocumentDetail(documentId string) (sdktypes.PlusApiResultDocumentDetailVO, error) {
    raw, err := a.client.Get(AppApiPath(fmt.Sprintf("/documents/%s", documentId)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentDetailVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentDetailVO](raw)
}

// Update document metadata
func (a *DocumentsApi) UpdateDocument(documentId string, body sdktypes.DocumentUpdateRequest) (sdktypes.PlusApiResultDocumentOperationVO, error) {
    raw, err := a.client.Put(AppApiPath(fmt.Sprintf("/documents/%s", documentId)), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentOperationVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentOperationVO](raw)
}

// Delete document
func (a *DocumentsApi) DeleteDocument(documentId string) (sdktypes.PlusApiResultVoid, error) {
    raw, err := a.client.Delete(AppApiPath(fmt.Sprintf("/documents/%s", documentId)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultVoid
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultVoid](raw)
}

// Restore document
func (a *DocumentsApi) RestoreDocument(documentId string) (sdktypes.PlusApiResultDocumentOperationVO, error) {
    raw, err := a.client.Put(AppApiPath(fmt.Sprintf("/documents/%s/restore", documentId)), nil, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentOperationVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentOperationVO](raw)
}

// Get document content
func (a *DocumentsApi) GetDocumentContent(documentId string) (sdktypes.PlusApiResultDocumentContentVO, error) {
    raw, err := a.client.Get(AppApiPath(fmt.Sprintf("/documents/%s/content", documentId)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentContentVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentContentVO](raw)
}

// Update document content
func (a *DocumentsApi) UpdateDocumentContent(documentId string, body sdktypes.DocumentContentUpdateRequest) (sdktypes.PlusApiResultDocumentContentVO, error) {
    raw, err := a.client.Put(AppApiPath(fmt.Sprintf("/documents/%s/content", documentId)), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentContentVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentContentVO](raw)
}

// Archive document
func (a *DocumentsApi) ArchiveDocument(documentId string) (sdktypes.PlusApiResultDocumentOperationVO, error) {
    raw, err := a.client.Put(AppApiPath(fmt.Sprintf("/documents/%s/archive", documentId)), nil, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentOperationVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentOperationVO](raw)
}

// List documents
func (a *DocumentsApi) ListDocuments(query map[string]interface{}) (sdktypes.PlusApiResultPageDocumentVO, error) {
    raw, err := a.client.Get(AppApiPath("/documents"), query, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultPageDocumentVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPageDocumentVO](raw)
}

// Create document
func (a *DocumentsApi) CreateDocument(body sdktypes.DocumentCreateRequest) (sdktypes.PlusApiResultDocumentOperationVO, error) {
    raw, err := a.client.Post(AppApiPath("/documents"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentOperationVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentOperationVO](raw)
}

// Favorite document
func (a *DocumentsApi) FavoriteDocument(documentId string) (sdktypes.PlusApiResultDocumentOperationVO, error) {
    raw, err := a.client.Post(AppApiPath(fmt.Sprintf("/documents/%s/favorite", documentId)), nil, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentOperationVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentOperationVO](raw)
}

// Unfavorite document
func (a *DocumentsApi) UnfavoriteDocument(documentId string) (sdktypes.PlusApiResultDocumentOperationVO, error) {
    raw, err := a.client.Delete(AppApiPath(fmt.Sprintf("/documents/%s/favorite", documentId)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentOperationVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentOperationVO](raw)
}

// Copy document
func (a *DocumentsApi) CopyDocument(documentId string, body sdktypes.DocumentCopyRequest) (sdktypes.PlusApiResultDocumentOperationVO, error) {
    raw, err := a.client.Post(AppApiPath(fmt.Sprintf("/documents/%s/copy", documentId)), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentOperationVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentOperationVO](raw)
}

// Batch update document
func (a *DocumentsApi) BatchUpdateDocument(documentId string, body sdktypes.DocumentBatchUpdateRequest) (sdktypes.PlusApiResultDocumentBatchUpdateResultVO, error) {
    raw, err := a.client.Post(AppApiPath(fmt.Sprintf("/documents/%s/batch-update", documentId)), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentBatchUpdateResultVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentBatchUpdateResultVO](raw)
}

// Batch update document
func (a *DocumentsApi) CreateBatchUpdateDocument(documentId string, body sdktypes.DocumentBatchUpdateRequest) (sdktypes.PlusApiResultDocumentBatchUpdateResultVO, error) {
    raw, err := a.client.Post(AppApiPath(fmt.Sprintf("/documents/%s:batchUpdate", documentId)), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentBatchUpdateResultVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentBatchUpdateResultVO](raw)
}

// Batch delete documents
func (a *DocumentsApi) BatchDelete() (sdktypes.PlusApiResultDocumentOperationVO, error) {
    raw, err := a.client.Delete(AppApiPath("/documents/batch-delete"), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentOperationVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentOperationVO](raw)
}

// Batch delete documents
func (a *DocumentsApi) DeleteBatch() (sdktypes.PlusApiResultDocumentOperationVO, error) {
    raw, err := a.client.Delete(AppApiPath("/documents/batch"), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentOperationVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentOperationVO](raw)
}

File: sdkwork-app-sdk-typescript/src/api/documents.ts
import { appApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { DocumentBatchUpdateRequest, DocumentContentUpdateRequest, DocumentCopyRequest, DocumentCreateRequest, DocumentUpdateRequest, PlusApiResultDocumentBatchUpdateResultVO, PlusApiResultDocumentContentVO, PlusApiResultDocumentDetailVO, PlusApiResultDocumentOperationVO, PlusApiResultPageDocumentVO, PlusApiResultVoid } from '../types';


export class DocumentsApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Get document detail */
  async getDocumentDetail(documentId: string | number): Promise<PlusApiResultDocumentDetailVO> {
    return this.client.get<PlusApiResultDocumentDetailVO>(appApiPath(`/documents/${documentId}`));
  }

/** Update document metadata */
  async updateDocument(documentId: string | number, body: DocumentUpdateRequest): Promise<PlusApiResultDocumentOperationVO> {
    return this.client.put<PlusApiResultDocumentOperationVO>(appApiPath(`/documents/${documentId}`), body);
  }

/** Delete document */
  async deleteDocument(documentId: string | number): Promise<PlusApiResultVoid> {
    return this.client.delete<PlusApiResultVoid>(appApiPath(`/documents/${documentId}`));
  }

/** Restore document */
  async restoreDocument(documentId: string | number): Promise<PlusApiResultDocumentOperationVO> {
    return this.client.put<PlusApiResultDocumentOperationVO>(appApiPath(`/documents/${documentId}/restore`));
  }

/** Get document content */
  async getDocumentContent(documentId: string | number): Promise<PlusApiResultDocumentContentVO> {
    return this.client.get<PlusApiResultDocumentContentVO>(appApiPath(`/documents/${documentId}/content`));
  }

/** Update document content */
  async updateDocumentContent(documentId: string | number, body: DocumentContentUpdateRequest): Promise<PlusApiResultDocumentContentVO> {
    return this.client.put<PlusApiResultDocumentContentVO>(appApiPath(`/documents/${documentId}/content`), body);
  }

/** Archive document */
  async archiveDocument(documentId: string | number): Promise<PlusApiResultDocumentOperationVO> {
    return this.client.put<PlusApiResultDocumentOperationVO>(appApiPath(`/documents/${documentId}/archive`));
  }

/** List documents */
  async listDocuments(params?: QueryParams): Promise<PlusApiResultPageDocumentVO> {
    return this.client.get<PlusApiResultPageDocumentVO>(appApiPath(`/documents`), params);
  }

/** Create document */
  async createDocument(body: DocumentCreateRequest): Promise<PlusApiResultDocumentOperationVO> {
    return this.client.post<PlusApiResultDocumentOperationVO>(appApiPath(`/documents`), body);
  }

/** Favorite document */
  async favoriteDocument(documentId: string | number): Promise<PlusApiResultDocumentOperationVO> {
    return this.client.post<PlusApiResultDocumentOperationVO>(appApiPath(`/documents/${documentId}/favorite`));
  }

/** Unfavorite document */
  async unfavoriteDocument(documentId: string | number): Promise<PlusApiResultDocumentOperationVO> {
    return this.client.delete<PlusApiResultDocumentOperationVO>(appApiPath(`/documents/${documentId}/favorite`));
  }

/** Copy document */
  async copyDocument(documentId: string | number, body: DocumentCopyRequest): Promise<PlusApiResultDocumentOperationVO> {
    return this.client.post<PlusApiResultDocumentOperationVO>(appApiPath(`/documents/${documentId}/copy`), body);
  }

/** Batch update document */
  async batchUpdateDocument(documentId: string | number, body: DocumentBatchUpdateRequest): Promise<PlusApiResultDocumentBatchUpdateResultVO> {
    return this.client.post<PlusApiResultDocumentBatchUpdateResultVO>(appApiPath(`/documents/${documentId}/batch-update`), body);
  }

/** Batch update document */
  async createBatchUpdateDocument(documentId: string | number, body: DocumentBatchUpdateRequest): Promise<PlusApiResultDocumentBatchUpdateResultVO> {
    return this.client.post<PlusApiResultDocumentBatchUpdateResultVO>(appApiPath(`/documents/${documentId}:batchUpdate`), body);
  }

/** Batch delete documents */
  async batchDelete(): Promise<PlusApiResultDocumentOperationVO> {
    return this.client.delete<PlusApiResultDocumentOperationVO>(appApiPath(`/documents/batch-delete`));
  }

/** Batch delete documents */
  async deleteBatch(): Promise<PlusApiResultDocumentOperationVO> {
    return this.client.delete<PlusApiResultDocumentOperationVO>(appApiPath(`/documents/batch`));
  }
}

export function createDocumentsApi(client: HttpClient): DocumentsApi {
  return new DocumentsApi(client);
}
